ruby/ext/google/protobuf_c/repeated_field.c

```diff
@@ -391,44 +375,6 @@ VALUE RepeatedField_eq(VALUE _self, VALUE _other) {
   return Qtrue;
 }
 
-/*
- * call-seq:
- *     RepeatedField.hash => hash_value
- *
- * Returns a hash value computed from this repeated field's elements.
- */
-VALUE RepeatedField_hash(VALUE _self) {
-  RepeatedField* self = ruby_to_RepeatedField(_self);
-
-  VALUE hash = LL2NUM(0);
-
-  upb_fieldtype_t field_type = self->field_type;
-  VALUE field_type_class = self->field_type_class;
-  size_t elem_size = native_slot_size(field_type);
-  size_t off = 0;
-  for (int i = 0; i < self->size; i++, off += elem_size) {
-    void* mem = ((uint8_t *)self->elements) + off;
-    VALUE elem = native_slot_get(field_type, field_type_class, mem);
-    hash = rb_funcall(hash, rb_intern("<<"), 1, INT2NUM(2));
-    hash = rb_funcall(hash, rb_intern("^"), 1,
-                      rb_funcall(elem, rb_intern("hash"), 0));
-  }
-
-  return hash;
-}
```

Comment:
If I'm understanding your data right, you're saying that `hash` on the repeated field with a Ruby-based implementation is half the speed of the fully-native C version, but still 25%-ish faster than a native Ruby array? Definitely interesting.

I'm not a huge fan of removing the `hash` implementations since they already exist (i.e., it's not saving work), and as far as I can tell, the value returned doesn't need to match what a native array would return (well, I suppose I could see both sides here, since `RepeatedField(:int32, [1, 2, 3]) == [1, 2, 3]`, but I'm not totally convinced that replying on this for a hashmap key is a good idea).

But this is a first-order proxy for what we may expect at least with the readonly `Array` methods, I suppose.